pkg/controller/ps/status.go

```diff
@@ -26,17 +27,33 @@ import (
 	"github.com/percona/percona-server-mysql-operator/pkg/router"
 )
 
-func (r *PerconaServerMySQLReconciler) reconcileCRStatus(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
-	if cr == nil || cr.ObjectMeta.DeletionTimestamp != nil {
-		return nil
-	}
-	if err := cr.CheckNSetDefaults(ctx, r.ServerVersion); err != nil {
-		cr.Status.State = apiv1alpha1.StateError
+func (r *PerconaServerMySQLReconciler) reconcileCRStatus(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL, reconcileErr error) error {
+	if reconcileErr != nil {
+		if cr.Status.State != apiv1alpha1.StateError {
+			meta.SetStatusCondition(&cr.Status.Conditions, metav1.Condition{
+				Status:             metav1.ConditionTrue,
+				Type:               string(apiv1alpha1.StateError),
+				Message:            reconcileErr.Error(),
+				Reason:             "ErrorReconcile",
+				LastTransitionTime: metav1.Now(),
+			})
+
+			cr.Status.Messages = append(cr.Status.Messages, "Error: "+reconcileErr.Error())
```

Comment:
if we keep appending errors maybe we should also add timestamps to each error? because it will be a long list of errors but users won't be able to understand what happened when

Reply:
This is a logic in PXC, but maybe we should. @hors wdyt